### ISSUE TITLE
[question] Best way to handle significant update for a package

### ISSUE THREAD

**Issue by ngrodzitski** [conan-io/conan-center-index#21030]
### What is your question?

Hi, 

I ask for advise on how to handle upcoming update for [restinio](https://github.com/Stiffstream/restinio) package.
From Conan perspective there would be the following changes:
* minimum cppstd: C++17
* Dependency for parser: `http_parser/2.9.4` => `llhttp/9.1.3` (hope https://github.com/conan-io/conan-center-index/pull/21024 will be accepted by the moment restinio release a new version).
* get rid of cmake-install for packaging (copy sources will be enough).

I'm thinking of moving current content of `recipes/restinio/all` to let's say `recipes/restinio/before-0.7.0` and modify all so that it accounts for `0.7.0+` versions only. And `recipes/restinio/config.yml` would be adjusted:
```
versions:
  "0.7.0":
    folder: all
  "0.6.19":
    folder: before-0.7.0
  "0.6.18":
    folder: before-0.7.0
  "0.6.17":
    folder: before-0.7.0
```

Handling new&old versions with same conanfile seems like a bad idea to me, because it would require many conditional handling.

If that is not sound way to do it, then could you, please, recommend how to approach that kind of package update?

**Comment by uilianries** [comment]
@ngrodzitski Thank you for asking! Indeed, we have some recipes in ConanCenterIndex that split the build folder according the version. The case that you are describing looks simple enough, I don't think it really needs a separated folder. It could be:

```python
@property
def _min_cppstd(self):
    return 14 if Version(self.version) < "0.7" else 17
    
@property
def _compilers_minimum_version(self):
    return {
        "apple-clang": "10",
        "clang": "3.4",
        "gcc": "5",
        "msvc": "190",
        "Visual Studio": "15",
    } if Version(self.version) < "0.7" else {
        "Visual Studio": "16",
        "msvc": "192",
        "gcc": "9",
        "clang": "9",
        "apple-clang": "12",
    }

def validate(self):
        if self.settings.get_safe("compiler.cppstd"):
            check_min_cppstd(self, self._min_cppstd)
        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
            raise ConanInvalidConfiguration(
                    f"{self.ref} requires C++{minimal_cpp_standard}, which your compiler does not support."
                )
```

**Comment by ngrodzitski** [comment]
@uilianries 
It's not quite so.

Dependencies are also different and In new version There would be no need for using cmake-install (copying header-files would be enough).

Here it is: https://github.com/conan-io/conan-center-index/pull/21098
Could you please check it?

**Comment by uilianries** [comment]
Fixed by #21098
